File: Python/1005. Maximize Sum Of Array After K Negations.py
'''
贪心算法：优先把最小的负数取正，如有剩余，对最小正数取反
'''
class Solution:
    def largestSumAfterKNegations(self, nums: List[int], k: int) -> int:
        nums = sorted(nums, key=abs, reverse=True)
        for i in range(len(nums)):
            if nums[i]<0 and k>0:
                nums[i] *= -1
                k -= 1
        if k%2 == 1:
            nums[-1] *= -1
        return sum(nums)

File: C++/1005. Maximize Sum Of Array After K Negations.cpp
/*
贪心算法：优先把最小的负数取正，如有剩余，对最小正数取反
*/
class Solution {
public:
    static bool cmp(int a, int b) {
        return abs(a) > abs(b);
    }
    int largestSumAfterKNegations(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end(), cmp);
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] < 0 && k>0) {
                nums[i] *= -1;
                k -= 1;
            }
        }
        if (k % 2 == 1) nums[nums.size() - 1] *= -1;
        int res = 0;
        for (int a : nums) res += a;
        return res;
    }
};